src/peft/utils/other.py

```diff
@@ -1556,3 +1554,57 @@ def create_attention_mask(
             position_ids=position_ids,
         )
     return attention_mask
+
+
+def _get_module_names_tied_with_embedding(model) -> list[str]:
+    """
+    Get the list of the fully qualified names of the modules that are tied to the input embeddings. In case of a
+    source-target-mapping `_tied_weights_keys`, it will attempt to identify the input embedding weights from the
+    mapping and return the list of tied modules accordingly. This gives a unified interface to both transformers v4
+    tied weights and v5 mapped tied weights.
```

Comment:
Perhaps worth mentioning:

- If the model is a PEFT model, it will be unpacked
- If the model is not a transformers model (but e.g. a custom model that wraps a transformers model), it should expose the `_tied_weights_keys` attribute.